File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int maxDiff = 0;
        
        for(int i = 0; i < prices.length; i++){
            int diff = prices[i] - min;
            
            if(diff >  maxDiff){
                maxDiff = diff;
            }
            
            if(prices[i] < min){
                min = prices[i];
            }
        }
        return maxDiff;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:    
    def maxProfit(self, prices: List[int]) -> int:
        minimum_so_far = prices[0]
        max_diff = 0
        
        for i in range(len(prices)):
            if minimum_so_far > prices[i]:
                minimum_so_far = prices[i]
            if prices[i] - minimum_so_far > max_diff:
                max_diff = prices[i] - minimum_so_far
        return max_diff